fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/LoanApplicationCommandFromApiJsonHelper.java

```diff
@@ -396,6 +396,12 @@ public void validateForCreate(final String json, final boolean isMeetingMandator
                 .extractStringNamed(LoanApiConstants.transactionProcessingStrategyCodeParameterName, element);
         baseDataValidator.reset().parameter(LoanApiConstants.transactionProcessingStrategyCodeParameterName)
                 .value(transactionProcessingStrategy).notNull();
+
+        if (!transactionProcessingStrategy.equals(loanProduct.getTransactionProcessingStrategyCode())) {
```

Comment:
This is not correct. You can change the transaction processing strategy before you are submitting a loan, except the "Advanced payment allocation" which is required to be set on loan product properly!